pkg/kube/snapshot/snapshot_alpha.go

```diff
@@ -402,14 +409,18 @@ func UnstructuredVolumeSnapshotContentAlpha(name, snapshotName, snapshotNs, dele
 				},
 				"volumeSnapshotRef": map[string]interface{}{
 					"kind":      VolSnapKind,
-					"name":      snapshotName,
-					"namespace": snapshotNs,
+					"name":      snapshotMeta.Name,
+					"namespace": snapshotMeta.Namespace,
 				},
 				"snapshotClassName": snapClassName,
 				"deletionPolicy":    deletionPolicy,
 			},
 		},
 	}
+	if contentMeta.Annotations != nil {
```

Comment:
Should we also set labels?

Reply:
Initially labels was not set, that's why didn't added as apart of this PR. We can take it up on another PR.